From b85b4f30846bb169c114e99ceee17cc119f02a4b Mon Sep 17 00:00:00 2001
From: Jason Gunthorpe <jgg@nvidia.com>
Date: Wed, 4 Oct 2023 09:08:32 -0300
Subject: [PATCH 2/2] iommu: Fix return code in
 iommu_group_alloc_default_domain()

This function returns NULL on errors, not ERR_PTR.

Fixes: 1c68cbc64fe6 ("iommu: Add IOMMU_DOMAIN_PLATFORM")
Reported-by: Dan Carpenter <dan.carpenter@linaro.org>
Link: https://lore.kernel.org/r/8fb75157-6c81-4a9c-9992-d73d49902fa8@moroto.mountain
Signed-off-by: Jason Gunthorpe <jgg@nvidia.com>
Link: https://lore.kernel.org/r/0-v2-ee2bae9af0f2+96-iommu_ga_err_ptr_jgg@nvidia.com
Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 drivers/iommu/iommu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -1787,7 +1787,7 @@ iommu_group_alloc_default_domain(struct
 	 */
 	if (ops->default_domain) {
 		if (req_type)
-			return ERR_PTR(-EINVAL);
+			return NULL;
 		return ops->default_domain;
 	}
 
